### PR TITLE
Bootloader Interface Fixes, Enhancements and Documentation

### PR DESCRIPTION
**Pull request by ejoerns**
This is a collection of several patches related to the bootchooser.c code as a result of originally only trying to add `BOOT_ORDER` handling to the U-Boot implementation.

As some of the design decisions of bootloader interaction are not that intuitive, I decided to add a new section to the reference part of the RAUC documentation for at least bringing a little light into the darkness.

The return handling fixes are part of the rework I promised (but still owe) @michaelolbrich when he acknowledged the g_autoptr patches of @pH5.

### CODE REVIEW

**Comment by michaelolbrich** [comment]
the cleanup stuff looks good.

**Comment by ejoerns** [comment]
@jluebbe  They look fine for me, thanks for review!

I've squashed the fixup commit and rebased my patches. Also made a new round of self-review targeting the documentation part. The fixes are attached as a new fixup commit.

File: docs/reference.rst [hunk 1]
```diff
@@ -785,3 +791,8 @@ EFI
 :primary:
   Sets the slot as `BootNext`.
   This will make the slot being booted upon next reboot only!
+
+.. note:: EFI implementations differ in how they handle new or unbootable
+  targets etc. It may also depend on the actual implementation if variabel
```
Comment (jluebbe) [inline docs/reference.rst]: s/variabel/EFI variable/

File: docs/reference.rst [hunk 2]
```diff
@@ -785,3 +791,8 @@ EFI
...
+  writing is atomic or not.
+  Thus be sure your EFI works as expected and required.
```
Comment (jluebbe) [inline docs/reference.rst]: s/be/make/

File: docs/reference.rst [hunk 3]
```diff
@@ -684,21 +684,25 @@ bootloader will not select it for booting anymore.
 As shown above this is either the case before an installation to make the
 update atomic from the bootloader's perspective, or optionally after the
 installation and a reboot into the new system, when a service detects that the
-system is in an unusable state. This potentially allows falling back to a
+system is in an unusable state. This potentially allows falling back to another
```
Comment (jluebbe) [inline docs/reference.rst]: I'd leave it as 'a', because we probably only have one working system in that situation.

**Comment by ejoerns** [comment]
I've changed the 3 items you mentioned and squashed the result together. Here it is.

**Pull request merged**